draft-ietf-httpbis-connect-tcp.md

```diff
@@ -44,9 +44,9 @@ HTTP/3 uses a UDP transport, so it cannot be forwarded using the pre-existing CO
 
 ## Problems
 
-HTTP proxies are usually identified by an "http" or "https" URI ({{?RFC9110, Section 7.3.2}}).  However, HTTP requests do not convey any of the proxy's URI components.  Instead, they only carry the URI of the request target (i.e., the origin) or, in the case of CONNECT requests, the host and port of the target.  This prevents any HTTP server from hosting multiple distinct proxy services.
+HTTP proxies are usually identified by an "http" or "https" URI ({{?RFC9110, Section 7.3.2}}).  However, HTTP requests using the proxy do not include any information from this URI.  Instead, they only indicate the URI of the request target (i.e., the origin) or, in the case of CONNECT requests, the host and port of the target.  This prevents any HTTP server from hosting multiple distinct proxy services, as the server cannot distinguish them by path (as with distinct resources) or by origin (as in "virtual hosting").
```

Comment:
This text in 9110 is about proxies in general, but doesn't apply to CONNECT proxies. When sending a GET request to a proxy, the scheme is present - but not for CONNECT. I think we were better off with the previous wording.

Reply:
GET requests to an HTTP request proxy carry the scheme of the target origin, not the proxy.  I don't see any indication that the text in 9110 is about request proxies but excludes CONNECT proxies, especially since CONNECT was always designed to be used alongside HTTP request proxying with the same proxy server (for "https" and "http" target origins, respectively).

Reply:
Section 7.3.2 of 9110 (and 7.3 more broadly) are specifically about routing _requests_. CONNECT is a means to establish a connection, not a way to send requests. From that section:
> the client connects inbound by establishing (or reusing) a connection to that proxy and then sending it an HTTP request message containing a request target that matches the client's target URI.

In the use-case of HTTPS-over-TLS-over-CONNECT-over-HTTP, the request message to the CONNECT proxy contains a request target that does *not* match the end-to-end target URI. So this section doesn't apply to CONNECT.

But perhaps a more productive path forward would be to focus on the fact that regular CONNECT doesn't carry a scheme nor path, and make some assertions specific to templated CONNECT. We might not agree on the details of what 9110 says, but I think we all agree that (1) regular CONNECT has pitfalls and (2) we know how to fix them in templated CONNECT - so might be best to focus on those points

Reply:
@DavidSchinazi 
> So this section doesn't apply to CONNECT.

This is true, but I wonder if the fact requires any changes to the text being proposed.

FWIW, the text says that "(the requests) only indicate the URI of the request target (i.e., the origin) or, in the case of CONNECT requests, the host and port of the target," which is accurate for CONNECT.

I think we agree on the pitfalls, and this paragraph is trying specifically to explain one of the pitfalls (i.e., why virtual hosting is impossible).

Is it the case that the proposed text is hard to digest?

Reply:
I think it's just the first sentence that bugs me because it doesn't match my understanding of CONNECT. A minimal change could be:

```suggestion
HTTP clients can be configured to use proxies by selecting a proxy hostname, a port, and whether to use a security protocol. However, CONNECT requests using the proxy do not carry this configuration information. Instead, they only indicate the hostname and port of the target. This prevents any HTTP server from hosting multiple distinct proxy services, as the server cannot distinguish them by path (as with distinct resources) or by origin (as in "virtual hosting").
```

Reply:
@DavidSchinazi Yeah, I get the confusion. While people may repurpose URIs for specifying proxies, it is questionable if everyone is used to thinking that way. In that sense I like your text.